java-components/build-request-processor/src/main/java/com/redhat/hacbs/container/analyser/build/gradle/GradleUtils.java

```diff
@@ -25,72 +18,28 @@ public final class GradleUtils {
     /**
      * Identifier for the plugin {@code com.github.sherter.google-java-format}.
      */
-    public static final String GOOGLE_JAVA_FORMAT_PLUGIN = Pattern.quote("com.github.sherter.google-java-format");
+    public static final String GOOGLE_JAVA_FORMAT_PLUGIN = "com.github.sherter.googlejavaformatgradleplugin.GoogleJavaFormatPlugin";
 
     static final String BUILD_GRADLE = "build.gradle";
 
     static final String BUILD_GRADLE_KTS = "build.gradle.kts";
 
-    private static final String GRADLE_WRAPPER_PROPERTIES = "gradle/wrapper/gradle-wrapper.properties";
-
-    private static final String DISTRIBUTION_URL_KEY = "distributionUrl";
-
-    private static final Pattern GRADLE_7_3_PLUS = Pattern.compile("^7\\.[3456789].*$");
-
-    private static final Pattern GRADLE_5_THROUGH_7 = Pattern.compile("^[567]\\..*$");
-
-    private static final Pattern DISTRIBUTION_URL_PATTERN = Pattern.compile("^.*/gradle-(?<version>.*)-(all|bin)\\.zip$");
-
-    private static final Pattern VERSION_PATTERN = Pattern
-            .compile("((\\d+)(\\.\\d+)+)(-(\\p{Alpha}+)-(\\w+))?(-(SNAPSHOT|\\d{14}([-+]\\d{4})?))?");
-
     private GradleUtils() {
 
     }
 
-    /**
-     * Gets the location of the {@code gradle-wrapper.properties} starting from the base directory
-     *
-     * @param basedir the base directory
-     * @return the path to {@code gradle-wrapper.properties}
-     */
-    public static Path getPropertiesFile(Path basedir) {
-        Path path = basedir.resolve(GRADLE_WRAPPER_PROPERTIES);
-        Log.infof("Returning path to gradle/wrapper/gradle-wrapper.properties: %s", path);
-        return path;
-    }
+    public static int getMinimumSupportedJavaVersion(String gradleVersion) {
+        GradleVersion version = GradleVersion.version(gradleVersion);
 
-    /**
-     * Get the Gradle version from {@code gradle/wrapper/gradle-wrapper.properties}.
-     *
-     * @param propertiesFile the location of {@code gradle/wrapper/gradle-wrapper.properties}
-     * @return the Gradle version
-     */
-    public static Optional<String> getGradleVersionFromWrapperProperties(Path propertiesFile) {
-        try (Reader reader = Files.newBufferedReader(propertiesFile)) {
-            var properties = new Properties();
-            properties.load(reader);
-            return getGradleVersionFromWrapperProperties(properties);
-        } catch (IOException e) {
-            Log.errorf("Error reading gradle-wrapper.properties: %s", e.getMessage(), e);
-            return Optional.empty();
+        if (version.compareTo(GradleVersion.version("4.0")) < 0) {
+            throw new IllegalArgumentException("Unsupported Gradle version: " + gradleVersion);
```

Comment:
Does this really need to throw an exception? If there is any chance projects using these old versions will build we should still attempt it with our oldest supported version.

Reply:
If it is a simple project it might still build with newer version though right? I am not sure how much the build scripts changed between those versions. 

Either way there is no harm in trying.